src/systemTest/java/com/flightstats/hub/callback/CallbackResource.java

```diff
@@ -13,23 +14,39 @@
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
 
 @Slf4j
 @Path("/callback")
 public class CallbackResource {
 
     private CallbackCache callbackCache;
+    private List<Predicate<WebhookCallbackRequest>> responseOverrides;
 
     @Inject
     public CallbackResource(CallbackCache callbackCache) {
         this.callbackCache = callbackCache;
+        this.responseOverrides = Collections.synchronizedList(new ArrayList<>());
```

Comment:
If thread safe collections are important, I highly recommend using java.util.concurrent classes. In this case, CopyOnWriteArrayList.